### PR TITLE
TRACE workflow improvements and build demand fixes

### PR DESCRIPTION
**Pull request by ollie-bell**
# Closes # (if applicable).

## Changes proposed in this Pull Request

* refactored and tidied up the run script. No longer calls a subprocess to submit batch jobs (still uses a subprocess for locally calling snakemake)
* added a script to check all workflow output files against what should be present
* fixed early termination after cutout has been built
* incorporated upstream fixes to building demand profiles and improved plexos missing demand fallback

## Checklist

- [ ] I consent to the release of this PR's code under the AGPLv3 license and non-code contributions under CC0-1.0 and CC-BY-4.0.
- [ ] I tested my contribution locally and it seems to work fine.
- [ ] Code and workflow changes are sufficiently documented.
- [ ] Newly introduced dependencies are added to `envs/environment.yaml` and `doc/requirements.txt`.
- [ ] Changes in configuration options are added in all of `config.default.yaml` and `config.tutorial.yaml`.
- [ ] Add a test config or line additions to `test/` (note tests are changing the config.tutorial.yaml)
- [ ] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [ ] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes, including reference to the requested PR.

### CODE REVIEW

File: ClimateTRACE/README.md [hunk 1]
```diff
@@ -5,21 +5,21 @@
 1. See the CLI help by running:
 
    ```bash
-   python ClimateTRACE/scripts/submit_job --help
+   python ClimateTRACE/run --help
```
Comment (ollie-bell) [inline ClimateTRACE/README.md:8]: the script to run the now workflow is now bundled separately from other adhoc scripts

File: ClimateTRACE/outputs/files.csv
Comment (ollie-bell) [inline ClimateTRACE/outputs/files.csv:1]: this is just a list of all outputs we should expect for a given run, just useful as a reference for now

File: ClimateTRACE/run/__main__.py [hunk 2]
```diff
@@ -136,7 +144,13 @@ def parse_args():
         nargs="+",
         help="Country ISO codes to EXCLUDE from config files",
     )
-    smk.add_argument("--weather-year", type=int, required=True, help="Model weather year to use.")
+    smk.add_argument(
+        "--weather-years",
+        nargs="+",
```
Comment (ollie-bell) [inline ClimateTRACE/run/__main__.py:149]: can now input multiple weather years to run

File: ClimateTRACE/scripts/verify_outputs.py
Comment (ollie-bell) [inline ClimateTRACE/scripts/verify_outputs.py:1]: script which checks outputs of a run against what to expect, still work in progress

File: Snakefile [hunk 3]
```diff
@@ -429,6 +429,17 @@ def terminate_if_cutout_exists(config=config):
...
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--mode", type=int, default=snakemake.Mode.default)
+    parser.add_argument("--target-jobs", type=str, nargs="+", default=[])
+    args, _ = parser.parse_known_args(sys.argv)
+    if args.mode == snakemake.Mode.subprocess and "build_cutout" not in " ".join(
+        args.target_jobs
+    ):
+        return
+
```
Comment (ollie-bell) [inline Snakefile:442]: this fixes the issue with the build cutout termination after it has already been built. I've made a PR upstream as well https://github.com/pypsa-meets-earth/pypsa-earth/pull/1079

File: configs/osm_config.yaml
Comment (ollie-bell) [inline configs/osm_config.yaml:1]: this was pulled in from upstream which fixed the issue with Russia demand not being found

File: scripts/build_demand_profiles.py [hunk 4]
```diff
@@ -62,6 +67,35 @@ def normed(s):
     return s / s.sum()
 
 
+def get_gegis_regions(countries):
```
Comment (ollie-bell) [inline scripts/build_demand_profiles.py:70]: from upstream

File: ClimateTRACE/run/gcp_utils.py [hunk 5]
```diff
@@ -74,19 +72,45 @@ def machine_compute_resource(*, machine_type: str, region: str, project_id: str)
     return cpu_milli, memory_mib
 
 
+def wait_for_jobs_to_succeed(batch_jobs: Sequence[batch.Job]):
```
Comment (Tomkourou) [inline ClimateTRACE/run/gcp_utils.py:75]: Can you explain the logic behind this?

Comment (ollie-bell) [reply ClimateTRACE/run/gcp_utils.py:75]: it was just moving stuff around which already existed. git thinks its new stuff but it isn't. We were never using this function but I kept it there anyway.

**Review by Tomkourou — APPROVED** [review]

**Pull request merged**